client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -98,16 +100,34 @@ public void channelCreated(Channel ch) throws Exception {
             }, channelPoolSize);
         }
 
-        private Single<HttpResponse> sendRequestInternalAsync(final HttpRequest request) {
+        private Single<HttpResponse> sendRequestInternalAsync(final HttpRequest request, Proxy proxy) {
             final URI uri;
             try {
-                uri = new URI(request.url());
+                if (proxy == null) {
+                    uri = new URI(request.url());
+                } else if (proxy.address() instanceof InetSocketAddress) {
+                    InetSocketAddress address = (InetSocketAddress) proxy.address();
+                    String scheme = address.getPort() == 443
+                            ? "https"
+                            : "http";
+
+                    String urlString = scheme + "://"
+                            + ((InetSocketAddress) proxy.address()).getHostString()
+                            + ":" + address.getPort();
+                    uri = new URI(urlString);
+                } else {
+                    throw new IllegalArgumentException(
+                            "SocketAddress on java.net.Proxy must be an InetSocketAddress. Instead found "
+                                    + proxy.address());
```

Comment:
You may want to write out the address's type in this exception too.

Reply:
Proxy.toString() might be suitable here--implementation appears to show both the type and address.